Sources/XiEditor/EditView.swift

```diff
@@ -561,6 +568,19 @@ final class EditView: NSView, NSTextInputClient, TextPlaneDelegate {
                 let x = dataSource.gutterWidth - (gutterXPad + CGFloat(gutterTL.width))
                 let y0 = yOff + dataSource.textMetrics.ascent + linespace * CGFloat(lineIx)
                 renderer.drawLine(line: gutterTL, x0: GLfloat(x), y0: GLfloat(y0))
+                
+                previousGutterNumber = gutterNumber
+                previousX = x
+                previousY0 = y0
+            } else {
+                // The case when the cursor is not at a logical line, but instead at a soft-wrapped line.
+                if line.containsCursor, let gutterNumber = previousGutterNumber, let x = previousX, let y0 = previousY0 {
+                    let gutterTL = gutterCache!.lookupLineNumber(lineIdx: gutterNumber, hasCursor: line.containsCursor)
```

Comment:
what if we move `let gutterTL = gutterCache!.lookupLineNumber(lineIdx: gutterNumber, hasCursor: line.containsCursor)` up to the `if let` clause?
That way we don't do a force-unwrap.
what do you think?

Reply:
I very much agree.  I went ahead and moved the unwrapping of gutterCache into the `if let` clause for both conditions, therefore removing both force-unwraps that were present.